sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -251,7 +253,11 @@ def send_options_request(max_retries=5):
           # This reports channel is READY but connections may fail
           # Seems to be only an issue on Mac with port forwardings
           return job_service.DescribePipelineOptions(
-              beam_job_api_pb2.DescribePipelineOptionsRequest())
+              beam_job_api_pb2.DescribePipelineOptionsRequest(),
+                  timeout=portable_options.job_server_timeout)
+        except grpc.FutureTimeoutError:
+          # no retry for timeout errors
```

Comment:
Are we sure we don't want retries? We could do exponential backoff, similar to what SubprocessJobServer already does.

Reply:
My guess is that we want to retry for other errors, but we should treat timeout as unretriable, so that we never wait for longer than the specified timeout (including retries). We could implement exponential backoff so that it will overall still wait for no longer than the timeout, but I'm not sure it's worth the additional complexity.

Reply:
Okay.